src/main/java/com/zufar/onlinestore/user/repository/UserRepository.java

```diff
@@ -2,12 +2,22 @@
 
 import com.zufar.onlinestore.user.entity.UserEntity;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
 import java.util.UUID;
 
+@Repository
 public interface UserRepository extends JpaRepository<UserEntity, UUID> {
 
     Optional<UserEntity> findByEmail(String email);
 
+    @Query(nativeQuery = true, value = """
```

Comment:
your query is straightforward so it does not require any database-specific features, it is preferable to use JPQL for its database portability and higher level of abstraction. Also u can remove @Param here cause its not mandatory